doc/rest_api/snippet_sharing/api-description.apib

```diff
@@ -0,0 +1,587 @@
+            + `application` - Application for which the configuration is meant
+            + `scope` - Application internal scope
+            + `slug` - Entry specific unique slug
+    + filter (string, optional) - Only entries containing the search string in either their path, title, description, author or tags will be added to the output.
+    + filterby (enum[string], optional) - On what fields the filter should be applied to (i.e. what fields to involve in search)
+        + Default: `all`
+        + Members
+            + `all` - All fields: key, title, description, author and tags
+            + `key` - Only the key
+            + `title` - Only the title
+            + `description` - Only the description
+            + `author` - Only the authors name
+            + `tags` - Only tags
+
++ Response 200 (application/json)
+    + Attributes (Entry List)
+
+
+## Get details for entry [GET /database/{organization}/{application}/{scope}/{slug}{?raw}]
+
+Returns detailed information about the entry with the specified key (path). If the optional parameter `raw` is supplied, a raw configuration is returned instead of the normal response.
+
++ Parameters
+    + organization (string, required) - The `organization` who built the `application` for which the configuration of the entry is meant
+    + application (string, required) - The `application` to which the configuration of the entry belongs to
+    + scope (string, required) - The `application` specific `scope` to which the configuration of the entry belongs to
+    + slug (string, required) - The unique identifier of the requested entry
+    + raw (string, optional) - If this parameter is supplied, the API will attempt to return the requested configuration snippet in the given `raw` format
+
++ Response 200 (application/json)
+    + Attributes (object)
+        + data (object) - The actual details about the entry
+            + key (object) - The key and its separate parts
+                + full (string) - The full key (path) of the entry
+                + organization (string) - The organization of the application to which the snippet belongs to
+                + application (string) - The application to which the snippet belongs to
+                + scope (string) - The application internal scope for the configuration snippet
+                + slug (string) - The unique identification slug of the entry
+            + meta (object) - Additional information about the entry
+                + title (string) - The descriptive title of the entry
+                + description (string) - The extended description of the entry
+                + author (string) - The name of the creator of the entry
+                + created_at (number) - A timestamp of the creation date of the entry
+            + value (array[object])
+                + format (string) - A configuration format, e.g. **ini**
```

Comment:
I had following ideas:
- When you fetch the details for a snippet, you get the snippet in all enabled configuration formats (plugins have to be enabled for use with the service first).
- Instead you can also fetch the snippet in a specific format directly by providing the name of the format. (This could be useful for use with curl and then pipeing it into a file or kdb for example.)

Although it doesn't really impact the API description, would it be better to read out all available storage plugins from kdb directly and use them instead of an individual setting? If I remember correctly, i invented this setting because the storage plugins do not behave consistently when they fail (or should fail) an import (e.g. the xml plugin).

About up- and download:
When creating a snippet, the submitted configuration snippet is run through all enabled plugins until one succeeds with the import (convertion into normal Elektra keys). It is then stored wherever the repository has been specified and in whatever format this file is mounted. 
Another solution would be to require the user to specify the format during upload and only use this plugin. Usability wise this seems not as good as the first approach to me, but definitely less error prone.